src/TorchSharp/Hub.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) .NET Foundation and Contributors.  All Rights Reserved.  See LICENSE in the project root for license information.
+using System;
+using System.IO;
+using System.Net;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace TorchSharp
+{
+    public static partial class torch
+    {
+        public static partial class hub
+        {
+            /// <summary>
+            /// Download the url to a file 
+            /// </summary>
+            /// <param name="url">The URL to download</param>
+            /// <param name="dst">The file path to download the URL into</param>
+            /// <param name="hash_prefix">If non null, the SHA256 hash of downloaded content must match this prefix</param>
+            /// <param name="progress">Display the progress bar</param>
+            /// <exception cref="InvalidDataException">SHA256 hash doesn't match</exception>
+            public static void download_url_to_file(string url, string dst, string hash_prefix = null, bool progress = true)
+            {
+                using (var webClient = new WebClient()) {
```

Comment:
>WebClient [](http://example.com/codeflow?start=43&length=9)

nit: WebClient is not recommended to use in any new development. Is it possible to use HttpClient here instead? 

[https://docs.microsoft.com/en-us/dotnet/api/system.net.webclient?view=net-6.0#remarks](https://docs.microsoft.com/en-us/dotnet/api/system.net.webclient?view=net-6.0#remarks)